be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -309,32 +339,68 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
             for (size_t i = 0; i < cur_pipeline_dop; ++i) {
                 auto&& operators = pipeline->create_operators(cur_pipeline_dop, i);
                 DriverPtr driver = std::make_shared<PipelineDriver>(std::move(operators), _query_ctx,
-                                                                    fragment_ctx.get(), driver_id++);
+                                                                    _fragment_ctx.get(), driver_id++);
                 setup_profile_hierarchy(pipeline, driver);
                 drivers.emplace_back(std::move(driver));
             }
         }
     }
     // The pipeline created later should be placed in the front
     runtime_state->runtime_profile()->reverse_childs();
-    fragment_ctx->set_drivers(std::move(drivers));
+    _fragment_ctx->set_drivers(std::move(drivers));
 
-    auto maybe_driver_token = exec_env->driver_limiter()->try_acquire(fragment_ctx->drivers().size());
+    auto maybe_driver_token = exec_env->driver_limiter()->try_acquire(_fragment_ctx->drivers().size());
     if (maybe_driver_token.ok()) {
-        fragment_ctx->set_driver_token(std::move(maybe_driver_token.value()));
+        _fragment_ctx->set_driver_token(std::move(maybe_driver_token.value()));
     } else {
         return maybe_driver_token.status();
     }
 
     if (wg != nullptr) {
-        for (auto& driver : fragment_ctx->drivers()) {
+        for (auto& driver : _fragment_ctx->drivers()) {
             driver->set_workgroup(wg);
         }
     }
 
-    _fragment_ctx = fragment_ctx.get();
-    _query_ctx->fragment_mgr()->register_ctx(fragment_instance_id, std::move(fragment_ctx));
+    return Status::OK();
+}
+
+Status FragmentExecutor::_prepare_global_dict(const TExecPlanFragmentParams& request) {
+    // Set up global dict
+    auto* runtime_state = _fragment_ctx->runtime_state();
+    if (request.fragment.__isset.query_global_dicts) {
+        RETURN_IF_ERROR(runtime_state->init_query_global_dict(request.fragment.query_global_dicts));
+    }
+
+    if (request.fragment.__isset.load_global_dicts) {
+        RETURN_IF_ERROR(runtime_state->init_load_global_dict(request.fragment.load_global_dicts));
+    }
+    return Status::OK();
+}
+
+Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParams& request) {
+    DCHECK(request.__isset.desc_tbl);
+    DCHECK(request.__isset.fragment);
+
+    bool prepare_success = false;
+    DeferOp defer([this, &prepare_success]() {
+        if (!prepare_success) {
+            _fail_cleanup();
+        }
+    });
+    RETURN_IF_ERROR(exec_env->query_pool_mem_tracker()->check_mem_limit("Start execute plan fragment."));
+
+    RETURN_IF_ERROR(_prepare_query_ctx(exec_env, request));
+    RETURN_IF_ERROR(_prepare_fragment_ctx(request));
+    ASSIGN_OR_RETURN(auto wg, _prepare_workgroup(request));
+    RETURN_IF_ERROR(_prepare_runtime_state(exec_env, request, wg));
```

Comment:
why not directly use `_wg`

Reply:
keep it to avoid destruct?

Reply:
Good advice, I could store the `shared_ptr<WorkGroup>` into `FragmentExecutor` to avoid keep a local variable.